### PR TITLE
DirectToLds for sgemm configurations

### PR DESCRIPTION
**Pull request by AlexBrownAMD**
Modify parameter check for DirectToLds for sgemm and add extended test suite with working configurations

### CODE REVIEW

File: Tensile/SolutionStructs.py [hunk 1]
```diff
@@ -2449,13 +2449,11 @@ def isDirectToLdsDoable(state, tc):
         reject(state, "can't use DirectToLds for not EnableMatrixInstruction and GlobalLoadVectorWidth%c * bpe * WavefrontSize > 256"%tc)
         return False
 
-    # TODO revisit fp32 case for failure
-    #if state["ProblemType"]["TLU%c"%tc] and numBytes < 8 and state["GlobalLoadVectorWidth%c"%tc] * numBytes > 4:
-    if numBytes < 8 and state["GlobalLoadVectorWidth%c"%tc] * numBytes > 4:
-      reject(state, "can't use DirectToLds for TLU%c and bpe < 8 and GlobalLoadVectorWidth%c * bpe > 4"%(tc, tc))
+    # NumLoadsCoalesced > 1 not working for sgemm
+    if numBytes < 8 and state["NumLoadsCoalesced%c"%tc] > 1:
```
Comment (nakajee) [inline Tensile/SolutionStructs.py:2453]: Probably better to unify this check with NumLoadsCoalesced + ZGEMM check below.

File: Tensile/Tests/extended/direct_to_lds/dtl_sgemm.yaml [hunk 2]
```diff
@@ -0,0 +1,1021 @@
...
+        - WaveSeparateGlobalReadB: [0]
+        - NumLoadsCoalescedA: [1] # NLC=2 not working
+        - NumLoadsCoalescedB: [1] # NLC=2 not working
+        - ScheduleIterAlg: [3]
+        - AssertSummationElementMultiple: [32]
+        - StaggerU: [0]
+        - NumElementsPerBatchStore: [0] # minimum 2 for SCIU
+        - FractionalLoad: [0, 1, 2]
+        - BufferLoad: [True, False]
+        - TransposeLDS: [0, 1]
```
Comment (nakajee) [inline Tensile/Tests/extended/direct_to_lds/dtl_sgemm.yaml:77]: TransposeLDS=0 should not be necessary for DTL test.
DTL works only with TransposeLDS=1 for TLU=false case.

File: Tensile/Tests/extended/direct_to_lds/dtl_sgemm.yaml [hunk 3]
```diff
@@ -0,0 +1,1021 @@
...
+        - WaveSeparateGlobalReadA: [0]
+        - WaveSeparateGlobalReadB: [0]
+        - NumLoadsCoalescedA: [1] # NLC=2 not working
+        - NumLoadsCoalescedB: [1] # NLC=2 not working
+        - ScheduleIterAlg: [3]
+        - AssertSummationElementMultiple: [32]
+        - StaggerU: [0]
+        - NumElementsPerBatchStore: [0] # minimum 2 for SCIU
+        - FractionalLoad: [0, 1, 2]
+        - BufferLoad: [True, False]
```
Comment (nakajee) [inline Tensile/Tests/extended/direct_to_lds/dtl_sgemm.yaml:76]: It looks like BufferLoad=False makes DirectToLds=False.
It does not verify DirectToLds.
It looks like flat_load does not support LDS flag for globalread.
(Table 44 in MI200 ISA manual)

**Comment by AlexBrownAMD** [comment]
Thanks for the review, updates made to address comments

**Review by nakajee — APPROVED** [review]
Thanks for your update.
I do not have further comments.

**Review by yoichiyoshida — APPROVED** [review]

**Pull request merged**